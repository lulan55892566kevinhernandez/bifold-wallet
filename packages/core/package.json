{
  "name": "@hyperledger/aries-bifold-core",
  "version": "1.0.0",
  "private": true,
  "main": "build/index",
  "types": "build/index",
  "files": [
    "build",
    "android",
    "ios",
    "cpp",
    "*.podspec",
    "!lib/typescript/example",
    "!ios/build",
    "!android/build",
    "!android/gradle",
    "!android/gradlew",
    "!android/gradlew.bat",
    "!android/local.properties",
    "!**/__tests__",
    "!**/__fixtures__",
    "!**/__mocks__",
    "!**/.*"
  ],
  "license": "Apache-2.0",
  "publishConfig": {
    "access": "public"
  },
  "homepage": "https://github.com/hyperledger/aries-mobile-agent-react-native/tree/main/packages/core",
  "repository": {
    "type": "git",
    "url": "https://github.com/hyperledger/aries-mobile-agent-react-native",
    "directory": "packages/core"
  },
  "scripts": {
    "build": "yarn run clean && yarn run compile",
    "clean": "rimraf ./build",
    "compile": "tsc -p tsconfig.build.json",
    "prepublishOnly": "yarn run build",
    "test": "jest"
  },
  "peerDependencies": {
    "react": "*",
    "react-native": "*"
  },
  "devDependencies": {
    "@react-native-community/eslint-config": "^3.2.0",
    "@types/jest": "^29.5.5",
    "@types/react": "^18.0.24",
    "@types/react-native": "0.71.5",
    "@types/ws": "^7.4.6",
    "react": "*",
    "react-native": "*",
    "rimraf": "^5.0.0",
    "typescript": "^5.0.4"
  }
}
