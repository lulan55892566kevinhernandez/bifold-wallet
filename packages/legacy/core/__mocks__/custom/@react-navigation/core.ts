const navigate = jest.fn()
const dispatch = jest.fn()
const replace = jest.fn()
const navigation = {
  __timestamp: process.hrtime(),
  navigate,
  replace,
  setOptions: jest.fn(),
  getParent: () => {return {
    navigate,
    dispatch,
    replace,
  }},
  getState: jest.fn(() => ({
    index: jest.fn(),
  })),
  goBack: jest.fn(),
  pop: jest.fn(),
  reset: jest.fn(),
  isFocused: () => true,
  dispatch,
}

const useNavigation = () => {
  return navigation
}

const useIsFocused = () => {
  return true
}

export { useNavigation, useIsFocused }
