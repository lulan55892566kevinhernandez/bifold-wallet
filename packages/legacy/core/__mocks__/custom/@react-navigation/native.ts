const useFocusEffect = jest.fn()
const createNavigatorFactory = jest.fn()

export { useFocusEffect, createNavigatorFactory }
