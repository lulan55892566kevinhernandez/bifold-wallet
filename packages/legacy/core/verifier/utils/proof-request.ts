import {
  Agent,
  AgentMessage,
  AutoAcceptProof,
  ProofExchangeRecord,
  ProofRequest,
  V1RequestPresentationMessage,
} from '@aries-framework/core'

import { defaultProofRequestTemplates } from '../constants'
import {
  IndyRequestedAttribute,
  IndyRequestedPredicate,
  ProofRequestTemplate,
  ProofRequestType,
} from '../types/proof-reqeust-template'

const protocolVersion = 'v1'
const domain = 'http://aries-mobile-agent.com'

/*
 * Find Proof Request message in the storage by the given id
 * */
export const findProofRequestMessage = async (agent: Agent, id: string): Promise<ProofRequest | undefined> => {
  const message = await agent.proofs.findRequestMessage(id)
  if (message && message instanceof V1RequestPresentationMessage && message.indyProofRequest) {
    return message.indyProofRequest
  } else {
    return undefined
  }
}

/*
 * Find Proof Request template for provided id
 * */
export const getProofRequestTemplate = (id: string): ProofRequestTemplate | undefined => {
  return defaultProofRequestTemplates.find((template) => template.id === id)
}

/*
 * Build Proof Request data from for provided template
 * */
export const buildProofRequestDataForTemplate = (
  template: ProofRequestTemplate,
  customValues?: Record<string, Record<string, number>>
) => {
  if (template.payload.type === ProofRequestType.Indy) {
    const requestedAttributes: Map<string, IndyRequestedAttribute> = new Map()
    const requestedPredicates: Map<string, IndyRequestedPredicate> = new Map()
    let index = 0

    template.payload.data.forEach((data) => {
      if (data.requestedAttributes?.length) {
        data.requestedAttributes.forEach((requestedAttribute) => {
          requestedAttributes.set(`referent_${index}`, requestedAttribute)
          index++
        })
      }
      if (data.requestedPredicates?.length) {
        data.requestedPredicates.forEach((requestedPredicate) => {
          const customValue =
            customValues && customValues[data.schema] ? customValues[data.schema][requestedPredicate.name] : undefined
          if (requestedPredicate.parameterizable && customValue) {
            requestedPredicates.set(`referent_${index}`, { ...requestedPredicate, predicateValue: customValue })
          } else {
            requestedPredicates.set(`referent_${index}`, requestedPredicate)
          }
          index++
        })
      }
    })
    return {
      indy: {
        name: template.name,
        version: template.version,
        nonce: Date.now().toString(),
        requestedAttributes,
        requestedPredicates,
      },
    }
  }
  if (template.payload.type === ProofRequestType.DIF) {
    return {}
  }
}

export const buildProofRequestDataForTemplateId = (
  templateId: string,
  customPredicateValues?: Record<string, Record<string, number>>
) => {
  const template = getProofRequestTemplate(templateId)
  if (!template) {
    return undefined
  }
  return buildProofRequestDataForTemplate(template, customPredicateValues)
}

export interface CreateProofRequestInvitationResult {
  request: AgentMessage
  proofRecord: ProofExchangeRecord
  invitation: AgentMessage
  invitationUrl: string
}

/*
 * Create connectionless proof request invitation for provided template
 * */
export const createConnectionlessProofRequestInvitation = async (
  agent: Agent,
  templateId: string,
  customPredicateValues?: Record<string, Record<string, number>>
): Promise<CreateProofRequestInvitationResult | undefined> => {
  const proofFormats = buildProofRequestDataForTemplateId(templateId, customPredicateValues)
  if (!proofFormats) {
    return undefined
  }
  const { message: request, proofRecord } = await agent.proofs.createRequest({
    protocolVersion,
    autoAcceptProof: AutoAcceptProof.Always,
    proofFormats,
  })
  const { message: invitation, invitationUrl } = await agent.oob.createLegacyConnectionlessInvitation({
    recordId: proofRecord.id,
    message: request,
    domain,
  })
  return {
    request,
    proofRecord,
    invitation,
    invitationUrl,
  }
}

export interface SendProofRequestResult {
  proofRecord: ProofExchangeRecord
}

/*
 * Build Proof Request for provided template and send it to provided connection
 * */
export const sendProofRequest = async (
  agent: Agent,
  templateId: string,
  connectionId: string,
  customPredicateValues?: Record<string, Record<string, number>>
): Promise<SendProofRequestResult | undefined> => {
  const proofFormats = buildProofRequestDataForTemplateId(templateId, customPredicateValues)
  if (!proofFormats) {
    return undefined
  }
  const proofRecord = await agent.proofs.requestProof({
    protocolVersion,
    connectionId,
    proofFormats,
  })
  return {
    proofRecord,
  }
}

/*
 * Check if the Proof Request template contains at least one predicate
 * */
export const hasPredicates = (record: ProofRequestTemplate): boolean => {
  if (record.payload.type === ProofRequestType.Indy) {
    return record.payload.data.some((d) => d.requestedPredicates && d.requestedPredicates?.length > 0)
  }
  if (record.payload.type === ProofRequestType.DIF) {
    return false
  }
  return false
}

/*
 * Check if the Proof Request template contains parameterizable predicates
 * */
export const isParameterizable = (record: ProofRequestTemplate): boolean => {
  if (record.payload.type === ProofRequestType.Indy) {
    return record.payload.data.some((d) => d.requestedPredicates?.some((predicate) => predicate.parameterizable))
  }
  if (record.payload.type === ProofRequestType.DIF) {
    return false
  }
  return false
}
