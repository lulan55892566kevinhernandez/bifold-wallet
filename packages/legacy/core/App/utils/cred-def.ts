import { CredentialMetadataKeys, CredentialExchangeRecord as CredentialRecord } from '@aries-framework/core'

import { credentialSchema, parseSchemaFromId } from './schema'

export function parseCredDefFromId(credDefId?: string, schemaId?: string): string {
  let name = 'Credential'
  if (credDefId) {
    const credDefRegex = /[^:]+/g
    const credDefIdParts = credDefId.match(credDefRegex)
    if (credDefIdParts?.length === 5) {
      name = `${credDefIdParts?.[4].replace(/_|-/g, ' ')}`
        .split(' ')
        .map((credIdPart) => credIdPart.charAt(0).toUpperCase() + credIdPart.substring(1))
        .join(' ')
    }
  }
  if (name.toLocaleLowerCase() === 'default' || name.toLowerCase() === 'credential') {
    name = parseSchemaFromId(schemaId).name
  }
  return name
}

function credentialDefinition(credential: CredentialRecord): string | undefined {
  return credential.metadata.get(CredentialMetadataKeys.IndyCredential)?.credentialDefinitionId
}

export function parsedCredDefName(credential: CredentialRecord): string {
  return parseCredDefFromId(credentialDefinition(credential), credentialSchema(credential))
}
