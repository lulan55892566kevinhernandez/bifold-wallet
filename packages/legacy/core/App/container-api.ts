import { StackNavigationProp } from '@react-navigation/stack'
import React, { createContext, useContext } from 'react'
import { DependencyContainer } from 'tsyringe'

import { Button } from './components/buttons/Button-api'
import { ReducerAction } from './contexts/reducers/store'
import Onboarding from './screens/Onboarding'
import { GenericFn } from './types/fn'
import { AuthenticateStackParams } from './types/navigators'

export enum SCREEN_TOKENS {
  SCREEN_PREFACE = 'screen.preface',
  SCREEN_TERMS = 'screen.terms',
  SCREEN_ONBOARDING = 'screen.onboarding',
  SCREEN_DEVELOPER = 'screen.developer',
  SCREEN_ONBOARDING_ITEM = 'screen.onboarding.item',
}
export enum STACK_TOKENS {
  STACK_ONBOARDING = 'stack.onboarding',
}
export enum FN_TOKENS {
  FN_ONBOARDING_DONE = 'fn.onboardingDone',
}

export enum COMP_TOKENS {
  COMP_BUTTON = 'comp.button',
}

export enum SERVICE_TOKENS {
  SERVICE_TERMS = 'screen.terms',
}

export const TOKENS = { ...SCREEN_TOKENS, ...SERVICE_TOKENS, ...STACK_TOKENS, ...FN_TOKENS, ...COMP_TOKENS }

export type FN_ONBOARDING_DONE = (
  dispatch: React.Dispatch<ReducerAction<unknown>>,
  navigation: StackNavigationProp<AuthenticateStackParams>
) => GenericFn

export interface TokenMapping {
  [TOKENS.SCREEN_PREFACE]: React.FC
  [TOKENS.STACK_ONBOARDING]: React.FC
  [TOKENS.SCREEN_TERMS]: { screen: React.FC; version: boolean | string }
  [TOKENS.SCREEN_DEVELOPER]: React.FC
  [TOKENS.SCREEN_ONBOARDING]: typeof Onboarding
  [TOKENS.FN_ONBOARDING_DONE]: FN_ONBOARDING_DONE
  [TOKENS.COMP_BUTTON]: Button
}

export interface Container {
  init(): Container
  resolve<K extends keyof TokenMapping>(token: K): TokenMapping[K]
  getContainer(): DependencyContainer
}

export const ContainerContext = createContext<Container | undefined>(undefined)

export const ContainerProvider = ContainerContext.Provider

// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
export const useContainer = () => useContext(ContainerContext)!
