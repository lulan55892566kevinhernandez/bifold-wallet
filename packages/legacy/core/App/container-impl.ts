import { StackNavigationProp } from '@react-navigation/stack'
import { createContext, useContext } from 'react'
import { DependencyContainer } from 'tsyringe'

import Button from './components/buttons/Button'
import { TOKENS, Container, TokenMapping } from './container-api'
import { DispatchAction, ReducerAction } from './contexts/reducers/store'
import OnboardingStack from './navigators/OnboardingStack'
import { DefaultScreenOptionsDictionary } from './navigators/defaultStackOptions'
import Developer from './screens/Developer'
import Onboarding from './screens/Onboarding'
import Preface from './screens/Preface'
import ScreenTerms, { TermsVersion } from './screens/Terms'
import { AuthenticateStackParams, Screens } from './types/navigators'

export class MainContainer implements Container {
  public static readonly TOKENS = TOKENS
  private container: DependencyContainer
  public constructor(container: DependencyContainer) {
    this.container = container
  }
  public init(): Container {
    // eslint-disable-next-line no-console
    console.log(`Initializing Bifold container`)
    this.container.registerInstance(TOKENS.SCREEN_PREFACE, Preface)
    this.container.registerInstance(TOKENS.SCREEN_DEVELOPER, Developer)
    this.container.registerInstance(TOKENS.SCREEN_TERMS, { screen: ScreenTerms, version: TermsVersion })
    this.container.registerInstance(TOKENS.SCREEN_ONBOARDING, Onboarding)
    this.container.registerInstance(TOKENS.STACK_ONBOARDING, OnboardingStack)
    this.container.registerInstance(TOKENS.COMP_BUTTON, Button)
    this.container.registerInstance(TOKENS.OBJECT_ONBOARDINGCONFIG, DefaultScreenOptionsDictionary)

    this.container.registerInstance(
      TOKENS.FN_ONBOARDING_DONE,
      (dispatch: React.Dispatch<ReducerAction<unknown>>, navigation: StackNavigationProp<AuthenticateStackParams>) => {
        return () => {
          dispatch({
            type: DispatchAction.DID_COMPLETE_TUTORIAL,
          })

          navigation.navigate(Screens.Terms)
        }
      }
    )

    return this
  }

  public resolve<K extends keyof TokenMapping>(token: K): TokenMapping[K] {
    // eslint-disable-next-line no-console
    console.log(`resolving ${token}`)
    return this.container.resolve(token) as TokenMapping[K]
  }

  public getContainer(): DependencyContainer {
    return this.container
  }
}

export const SystemContext = createContext<Container | undefined>(undefined)

export const SystemProvider = SystemContext.Provider

// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
export const useSystem = () => useContext(SystemContext)!
