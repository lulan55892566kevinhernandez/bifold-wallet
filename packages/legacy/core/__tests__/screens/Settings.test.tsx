import { useNavigation } from '@react-navigation/core'
import { render } from '@testing-library/react-native'
import React from 'react'

import Settings from '../../App/screens/Settings'
import configurationContext from '../contexts/configuration'
import { ConfigurationContext } from '../../App/contexts/configuration'
import { testIdWithKey } from '../../App/utils/testable'
import { StoreContext, StoreProvider } from '../../App'
import { testDefaultState } from '../contexts/store'

jest.mock('@react-navigation/core', () => {
  return require('../../__mocks__/custom/@react-navigation/core')
})
jest.mock('@react-navigation/native', () => {
  return require('../../__mocks__/custom/@react-navigation/native')
})
jest.mock('react-native-fs', () => ({}))
jest.mock('@hyperledger/anoncreds-react-native', () => ({}))
jest.mock('@hyperledger/aries-askar-react-native', () => ({}))
jest.mock('@hyperledger/indy-vdr-react-native', () => ({}))
jest.mock('react-native-permissions', () => require('react-native-permissions/mock'))

jest.mock('react-native-device-info', () => {
  return {
    getVersion: () => 1,
    getBuildNumber: () => 1,
  }
})

describe('Settings Screen', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  test('Renders correctly', async () => {
    const tree = render(
      <ConfigurationContext.Provider value={configurationContext}>
        <Settings navigation={useNavigation()} route={{} as any} />
      </ConfigurationContext.Provider>
    )
    expect(tree).toMatchSnapshot()
  })

  test('If developer mode is enabled, developer mode button is shown', async () => {
    const customState = {
      ...testDefaultState,
      preferences: {
        ...testDefaultState.preferences,
        developerModeEnabled: true
      }
    } 
    const tree = render(
      <StoreContext.Provider value={[customState, () => { return }]}>
        <ConfigurationContext.Provider value={configurationContext}>
          <Settings navigation={useNavigation()} route={{} as any} />
        </ConfigurationContext.Provider>
      </StoreContext.Provider>
    )
    
    const developerModeButton = tree.getByTestId(testIdWithKey('DeveloperOptions'))
    expect(developerModeButton).not.toBeNull()    
  })

  test('If mobile verifier is enabled, verifier options are shown', async () => {
    const customState = {
      ...testDefaultState,
      preferences: {
        ...testDefaultState.preferences,
        useVerifierCapability: true
      }
    }    
    const tree = render(
      <StoreContext.Provider value={[customState, () => { return }]}>
        <ConfigurationContext.Provider value={configurationContext}>
          <Settings navigation={useNavigation()} route={{} as any} />
        </ConfigurationContext.Provider>
      </StoreContext.Provider>
    )
    const proofButton = tree.getByTestId(testIdWithKey('ProofRequests'))
    expect(proofButton).not.toBeNull()
  })
})
