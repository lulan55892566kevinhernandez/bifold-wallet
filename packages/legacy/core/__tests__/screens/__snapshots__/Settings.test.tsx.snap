// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Screen Renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
    ]
  }
  style={
    Object {
      "backgroundColor": "#000000",
      "width": "100%",
    }
  }
>
  <RCTScrollView
    ListFooterComponent={[Function]}
    data={
      Array [
        Object {
          "data": Array [
            Object {
              "accessibilityLabel": "Screens.Contacts",
              "onPress": [Function],
              "testID": "com.ariesbifold:id/Contacts",
              "title": "Screens.Contacts",
            },
            Object {
              "accessibilityLabel": "Settings.WhatAreContacts",
              "onPress": [Function],
              "testID": "com.ariesbifold:id/WhatAreContacts",
              "title": "Settings.WhatAreContacts",
              "value": undefined,
            },
          ],
          "header": Object {
            "icon": "apartment",
            "title": "Screens.Contacts",
          },
        },
        Object {
          "data": Array [
            Object {
              "accessibilityLabel": "Global.Biometrics",
              "onPress": [Function],
              "testID": "com.ariesbifold:id/Biometrics",
              "title": "Global.Biometrics",
              "value": "Global.Off",
            },
            Object {
              "accessibilityLabel": "Settings.ChangePin",
              "onPress": [Function],
              "testID": "com.ariesbifold:id/Change Pin",
              "title": "Settings.ChangePin",
              "value": undefined,
            },
            Object {
              "accessibilityLabel": "Settings.Language",
              "onPress": [Function],
              "testID": "com.ariesbifold:id/Language",
              "title": "Settings.Language",
              "value": "Language.English",
            },
          ],
          "header": Object {
            "icon": "settings",
            "title": "Settings.AppSettings",
          },
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#313132",
                "flexGrow": 1,
                "paddingVertical": 24,
              },
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "marginBottom": -11,
                "paddingBottom": 0,
                "paddingHorizontal": 25,
              },
            ]
          }
        >
          <Text
            accessible={false}
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 24,
                },
                Object {
                  "color": "#FFFFFF",
                  "marginRight": 10,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            î©€
          </Text>
          <Text
            accessibilityRole="header"
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 26,
                  "fontWeight": "bold",
                },
                Object {
                  "flexShrink": 1,
                },
              ]
            }
          >
            Screens.Contacts
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            style={
              Array [
                Object {
                  "marginBottom": 10,
                },
              ]
            }
          />
          <RCTScrollView
            contentContainerStyle={
              Object {
                "flexGrow": 1,
              }
            }
            horizontal={true}
            style={
              Object {
                "backgroundColor": "#313132",
                "flexGrow": 1,
                "paddingVertical": 24,
              }
            }
          >
            <View>
              <View
                accessibilityLabel="Screens.Contacts"
                accessible={true}
                collapsable={false}
                focusable={true}
                nativeID="animatedComponent"
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "justifyContent": "space-between",
                    "opacity": 1,
                    "paddingHorizontal": 25,
                  }
                }
                testID="com.ariesbifold:id/Contacts"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 21,
                        "fontWeight": "bold",
                      },
                      Object {
                        "fontWeight": "normal",
                        "marginRight": 14,
                      },
                    ]
                  }
                >
                  Screens.Contacts
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 21,
                        "fontWeight": "bold",
                      },
                      Object {
                        "color": "#FFFFFF",
                        "fontWeight": "normal",
                      },
                    ]
                  }
                />
              </View>
            </View>
          </RCTScrollView>
          <View
            style={
              Object {
                "backgroundColor": "#313132",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderBottomColor": "#000000",
                    "borderBottomWidth": 1,
                    "marginHorizontal": 25,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <RCTScrollView
            contentContainerStyle={
              Object {
                "flexGrow": 1,
              }
            }
            horizontal={true}
            style={
              Object {
                "backgroundColor": "#313132",
                "flexGrow": 1,
                "paddingVertical": 24,
              }
            }
          >
            <View>
              <View
                accessibilityLabel="Settings.WhatAreContacts"
                accessible={true}
                collapsable={false}
                focusable={true}
                nativeID="animatedComponent"
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "justifyContent": "space-between",
                    "opacity": 1,
                    "paddingHorizontal": 25,
                  }
                }
                testID="com.ariesbifold:id/WhatAreContacts"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 21,
                        "fontWeight": "bold",
                      },
                      Object {
                        "fontWeight": "normal",
                        "marginRight": 14,
                      },
                    ]
                  }
                >
                  Settings.WhatAreContacts
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 21,
                        "fontWeight": "bold",
                      },
                      Object {
                        "color": "#FFFFFF",
                        "fontWeight": "normal",
                      },
                    ]
                  }
                />
              </View>
            </View>
          </RCTScrollView>
          <View
            style={
              Array [
                Object {
                  "marginBottom": 10,
                },
              ]
            }
          />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#313132",
                "flexGrow": 1,
                "paddingVertical": 24,
              },
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "marginBottom": -11,
                "paddingBottom": 0,
                "paddingHorizontal": 25,
              },
            ]
          }
        >
          <Text
            accessible={false}
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 24,
                },
                Object {
                  "color": "#FFFFFF",
                  "marginRight": 10,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            î¢¸
          </Text>
          <Text
            accessibilityRole="header"
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 26,
                  "fontWeight": "bold",
                },
                Object {
                  "flexShrink": 1,
                },
              ]
            }
          >
            Settings.AppSettings
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            style={
              Array [
                Object {
                  "marginBottom": 10,
                },
              ]
            }
          />
          <RCTScrollView
            contentContainerStyle={
              Object {
                "flexGrow": 1,
              }
            }
            horizontal={true}
            style={
              Object {
                "backgroundColor": "#313132",
                "flexGrow": 1,
                "paddingVertical": 24,
              }
            }
          >
            <View>
              <View
                accessibilityLabel="Global.Biometrics"
                accessible={true}
                collapsable={false}
                focusable={true}
                nativeID="animatedComponent"
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "justifyContent": "space-between",
                    "opacity": 1,
                    "paddingHorizontal": 25,
                  }
                }
                testID="com.ariesbifold:id/Biometrics"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 21,
                        "fontWeight": "bold",
                      },
                      Object {
                        "fontWeight": "normal",
                        "marginRight": 14,
                      },
                    ]
                  }
                >
                  Global.Biometrics
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 21,
                        "fontWeight": "bold",
                      },
                      Object {
                        "color": "#FFFFFF",
                        "fontWeight": "normal",
                      },
                    ]
                  }
                >
                  Global.Off
                </Text>
              </View>
            </View>
          </RCTScrollView>
          <View
            style={
              Object {
                "backgroundColor": "#313132",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderBottomColor": "#000000",
                    "borderBottomWidth": 1,
                    "marginHorizontal": 25,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <RCTScrollView
            contentContainerStyle={
              Object {
                "flexGrow": 1,
              }
            }
            horizontal={true}
            style={
              Object {
                "backgroundColor": "#313132",
                "flexGrow": 1,
                "paddingVertical": 24,
              }
            }
          >
            <View>
              <View
                accessibilityLabel="Settings.ChangePin"
                accessible={true}
                collapsable={false}
                focusable={true}
                nativeID="animatedComponent"
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "justifyContent": "space-between",
                    "opacity": 1,
                    "paddingHorizontal": 25,
                  }
                }
                testID="com.ariesbifold:id/Change Pin"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 21,
                        "fontWeight": "bold",
                      },
                      Object {
                        "fontWeight": "normal",
                        "marginRight": 14,
                      },
                    ]
                  }
                >
                  Settings.ChangePin
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 21,
                        "fontWeight": "bold",
                      },
                      Object {
                        "color": "#FFFFFF",
                        "fontWeight": "normal",
                      },
                    ]
                  }
                />
              </View>
            </View>
          </RCTScrollView>
          <View
            style={
              Object {
                "backgroundColor": "#313132",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderBottomColor": "#000000",
                    "borderBottomWidth": 1,
                    "marginHorizontal": 25,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <RCTScrollView
            contentContainerStyle={
              Object {
                "flexGrow": 1,
              }
            }
            horizontal={true}
            style={
              Object {
                "backgroundColor": "#313132",
                "flexGrow": 1,
                "paddingVertical": 24,
              }
            }
          >
            <View>
              <View
                accessibilityLabel="Settings.Language"
                accessible={true}
                collapsable={false}
                focusable={true}
                nativeID="animatedComponent"
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "justifyContent": "space-between",
                    "opacity": 1,
                    "paddingHorizontal": 25,
                  }
                }
                testID="com.ariesbifold:id/Language"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 21,
                        "fontWeight": "bold",
                      },
                      Object {
                        "fontWeight": "normal",
                        "marginRight": 14,
                      },
                    ]
                  }
                >
                  Settings.Language
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 21,
                        "fontWeight": "bold",
                      },
                      Object {
                        "color": "#FFFFFF",
                        "fontWeight": "normal",
                      },
                    ]
                  }
                >
                  Language.English
                </Text>
              </View>
            </View>
          </RCTScrollView>
          <View
            style={
              Array [
                Object {
                  "marginBottom": 10,
                },
              ]
            }
          />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "marginVertical": 25,
            }
          }
        >
          <View
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 18,
                  "fontWeight": "normal",
                }
              }
              testID="com.ariesbifold:id/Version"
            >
              Settings.Version 1 Settings.Build (1)
            </Text>
            <
              height={64}
              marginVertical={16}
              width="50%"
            />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
